libseam_carving/carver.cpp

```diff
@@ -25,33 +27,32 @@ namespace seam_carving {
             // loop through each row
             for (int row = 0; row < rows; row++) {
                 // add the current Coord to the seam
-                seam.pushCoord(Coord(row, col));
+                seam.push(Coord(row, col));
 
                 if (row != rows - 1) {
                     // find the direction of min and adjust path
-                    const ushort &center = energy_map.at<ushort>(row + 1, col);
-                    const ushort &left = (col - 1 < 0) ?
+                    const ushort& center = energy_map.at<ushort>(row + 1, col);
+                    const ushort& left = (col - 1 < 0) ?
                                          std::numeric_limits<ushort>::max() :
                                          energy_map.at<ushort>(row + 1, col - 1);
-                    const ushort &right = (col + 1 >= cols) ?
+                    const ushort& right = (col + 1 >= cols) ?
                                           std::numeric_limits<ushort>::max() :
                                           energy_map.at<ushort>(row + 1, col + 1);
 
                     // find the min pixel and adjust weight
-                    int min_energy = std::min({center, left, right});
-                    if (min_energy == left) col--;
-                    else if (min_energy == right) col++;
+                    if (left < center && left < right) col--;
```

Comment:
This is a little bit more verbose but tightens the bounds on this logic more than `std::min`. The goal is ultimately to determine which of `{center, left, right}` is the lowest value and if it's left or right, adjust the col. However, using `std::min` only returns the lowest value and doesn't offer any insight into which variable was the lowest. Thus, if `center == left` the same value is returned but it's unclear if the col should be adjusted or not. With the new syntax, it's explicitly that to adjust the col, left or right must be strictly lower than its counterparts.